blueocean-dashboard/src/main/js/components/CellLink.js

```diff
@@ -0,0 +1,42 @@
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+
+/**
+ * A convenience wrapper to pass down 'linkUrl' to child CellLink components.
+ */
+export class CellRow extends React.Component {
+
+    render() {
+        return (
+            <tr>
+                {React.Children.map(
+                    this.props.children,
+                    child => React.cloneElement(child, { linkUrl: this.props.linkUrl })
+                )}
+            </tr>
+        );
+    }
+}
+
+CellRow.propTypes = {
+    children: PropTypes.node,
+    linkUrl: PropTypes.string,
+};
+
+/**
+ * A table cell with an embedded react-router link.
+ */
+export function CellLink(props) {
+    const url = props.linkUrl || '';
+
+    return (
+        <td className="tableRowLink">
+            <Link to={url}>{props.children}</Link>
```

Comment:
that will blow up i18n or any feature that depends on ```query``` parameters I am afraid.

Have a look in src/main/js/components/PipelineRowItem.jsx

```
                    <Link to={activitiesURL} query={location.query}>
                        <ExpandablePath path={fullDisplayPath} />
                    </Link>
```